interfaces/Matlab_Toolbox_Revamp/Class/Kinetics.m

```diff
@@ -0,0 +1,590 @@
+            %    7])" returns a sparse matrix containing only the
+            %    coefficients for specis 3 in reactions 1, 3, 5, and 7. 
+                        
+            checklib;
+            nsp = kin.nSpecies;
+            nr = kin.nReactions;
+            temp = sparse(nsp, nr);
+            if nargin == 1
+                krange = 1:nsp;
+                irange = 1:nr;
+            elseif nargin == 3
+                krange = species;
+                irange = rxns;
+            else error('stoich_r requires 1 or 3 arguments.')
+            end
+            
+            for k = krange
+                for i = irange
+                    t = calllib(ct, 'kin_reactantStoichCoeff', ...
+                                kin.kin_id, k-1, i-1);
+                    if t ~= 0.0
+                        temp(k, i) = t;
+                    end
+                end
+            end
+            
+            n = temp;
+        end        
+        
+        function n = stoich_p(kin, species, rxns)
+            % Get the product stoichiometric coefficients. 
+            %
+            % parameter species:
+            %    Species indices for which product stoichiometric 
+            %    coefficients should be retrieved. Optional argument; if 
+            %    specified, "rxns" must be specified as well. 
+            % parameter rxns:
+            %    Reaction indicies for which product stoichiometric
+            %    coefficients should be retrieved. Optional argument; if
+            %    specified, "species" must be specified as well.
+            % return:
+            %    Returns a sparse matrix of all product stoichiometric
+            %    coefficients. The matrix elements "nu(k, i)" is the
+            %    stoichiometric coefficient of species k as a product in
+            %    reaction i. If "species" and "rxns" are specified, the
+            %    matrix will contain only entries for the specified species
+            %    and reactions. For example, "stoich_p(a, 3, [1, 3, 5,
+            %    7])" returns a sparse matrix containing only the
+            %    coefficients for specis 3 in reactions 1, 3, 5, and 7. 
+                        
+            checklib;
+            nsp = kin.nSpecies;
+            nr = kin.nReactions;
+            temp = sparse(nsp, nr);
+            if nargin == 1
+                krange = 1:nsp;
+                irange = 1:nr;
+            elseif nargin == 3
+                krange = species;
+                irange = rxns;
+            else error('stoich_p requires 1 or 3 arguments.')
+            end
+            
+            for k = krange
+                for i = irange
+                    t = calllib(ct, 'kin_productStoichCoeff', ...
+                                kin.kin_id, k-1, i-1);
+                    if t ~= 0.0
+                        temp(k, i) = t;
+                    end
+                end
+            end

```

Comment:
There is a new method in C++ (`Eigen::SparseMatrix<double> reactantStoichCoeffs() const`) that may allow for a significantly faster implementation as long as there's a suitable method exposing this in `clib`.

Also, this code segment has a couple of `^M` line endings that should be removed (here and many other instances).